src/content/posts/4-way-handshake.md/index.md

```diff
@@ -0,0 +1,139 @@
+**This request is not included in the 4-Way Handshake.**
+
+## 4-Way Handshake
+
+![schema](schema.png)
+
+This diagram shows the different requests and what they contain.
+
+> Note: *"STA" stands for Station, and "AP" for Access Point*  
+
+
+## First Request
+Firstly, the AP sends a nonce, which is a large random integer.
+
+When the station receives the packet, the station creates the PTK (Pairwise Transient Key) using the Key Derivation Function (KDF).
+This key is created with:
+1. The Pre-Shared Key (PSK),
+2. The ANonce received previously,
+3. The SNonce.
+
+The SNonce is similar to the ANonce, a large random number generated on the station side.
+Each time the station tries to authenticate, the SNonce and ANonce are generated again.
+These numbers ensure that the PTK is temporary.
+
+The KDF algorithm returns the PTK.
+We don’t use the password or the PSK directly because we don’t want the explicit key to go through the network.
+This is quite obvious; we don’t want others to easily intercept the password.
+
+### Pre-Shared Key (PSK)
+To get the PSK, the device only needs the SSID (network MAC address) and the password.  
+=> The process is really simple:  
+**Hash the password with the SSID as salt using the SHA-1 algorithm.**
+
+Now, we have the PSK (the AP also performs this operation).
+
+**The PSK is calculated before authentication starts (the station has all infos it needs).**  
+**The PMK is exactly the same as PSK for both WPA-Personal and WPA2-Personal**
+
+**VERY IMPORTANT**
+
+*For WPA-Entreprise and WPA2-Entreprise, they use here the PMK created by RADIUS authentication*
+
+## Second Request
+Let’s make a short recap.  
+At the beginning, we only know the **PSK=PMK**, which we derive from the **network password** and **SSID**.  
+*PMK is derived from RADIUS authentication in case of WPA-Entreprise and WPA2-Entreprise*  
+
+We received the AP's nonce from the first request, and the station generated the SNonce to create the **PTK** from the **ANonce**, **SNonce**, and **PSK**.
+
+Now we need to tell the AP that we know the password without explicitly sending the password.
+
+The station sends a packet containing the SNonce, allowing the AP to generate the same PTK.
+At the end of the packet, a **signature called the Message Integrity Code (MIC)** is added.
+  
+The MIC is just a SHA-1 hash of the packet content with the PTK as salt.  
+When the AP receives the second request, it will generate its PTK (it has all the necessary information: SNonce, ANonce, PSK).
+However, at this moment, the AP doesn't know if the station has the correct password.
+
+The AP generates the MIC from its newly generated PTK and compares it with the provided MIC.  
+**If the MICs are identical => PTKs are identical => PSKs are identical => passwords are identical**
+
+## Third Request
+To tell the station that it has the correct password, the AP won't do so explicitly.  
+**The AP will use the same strategy: a MIC.**
+

```

Comment:
Why ? real question what kind of attacks could happen otherwise

Reply:
good remark, I added a short answer